integrations/cdi/oci/src/main/java/io/helidon/integrations/cdi/oci/AdpSelectionStrategy.java

```diff
@@ -0,0 +1,545 @@
+                }
+            }
+            return false;
+        }
+
+        @Override // AdpSelectionStrategy
+        InstancePrincipalsAuthenticationDetailsProviderBuilder produceBuilder(Selector s, Config c, Annotation[] qs) {
+            return InstancePrincipalsAuthenticationDetailsProvider.builder();
+        }
+
+        @Override // AdpSelectionStrategy
+        InstancePrincipalsAuthenticationDetailsProvider produce(Selector selector, Config c, Annotation[] qualifiersArray) {
+            return selector.select(InstancePrincipalsAuthenticationDetailsProviderBuilder.class, qualifiersArray).get().build();
+        }
+    },
+
+    /**
+     * An {@link AdpSelectionStrategy} that works with {@link
+     * ResourcePrincipalAuthenticationDetailsProvider} instances.
+     */
+    RESOURCE_PRINCIPAL(ResourcePrincipalAuthenticationDetailsProvider.class,
+                       ResourcePrincipalAuthenticationDetailsProviderBuilder.class) {
+
+        @Override // AdpSelectionStrategy
+        boolean isAvailable(Selector selector, Config c, Annotation[] qualifiersArray) {
+            return
+                super.isAvailable(selector, c, qualifiersArray)
+                // https://github.com/oracle/oci-java-sdk/blob/v2.19.0/bmc-common/src/main/java/com/oracle/bmc/auth/ResourcePrincipalAuthenticationDetailsProvider.java#L246-L251
+                && System.getenv("OCI_RESOURCE_PRINCIPAL_VERSION") != null;
+        }
+
+        @Override // AdpSelectionStrategy
+        ResourcePrincipalAuthenticationDetailsProviderBuilder produceBuilder(Selector s, Config c, Annotation[] qualifiersArray) {
+            return ResourcePrincipalAuthenticationDetailsProvider.builder();
+        }
+
+        @Override // AdpSelectionStrategy
+        ResourcePrincipalAuthenticationDetailsProvider produce(Selector selector, Config c, Annotation[] qualifiersArray) {
+            return selector.select(ResourcePrincipalAuthenticationDetailsProviderBuilder.class, qualifiersArray).get().build();
+        }
+    },
+
+    /**
+     * An {@linkplain #isAbstract() abstract} {@link
+     * AdpSelectionStrategy} that selects the most appropriate
+     * concrete {@link AdpSelectionStrategy} by consulting
+     * configuration.
+     */
+    AUTO(AbstractAuthenticationDetailsProvider.class, true) {
+
+        private final transient Logger logger =
+            Logger.getLogger(this.getClass().getName(), AdpSelectionStrategy.class.getName() + "Messages");
+
+        @Override // AdpSelectionStrategy
+        Void produceBuilder(Selector selector, Config c, Annotation[] qualifiersArray) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override // AdpSelectionStrategy
+        AbstractAuthenticationDetailsProvider produce(Selector selector, Config c, Annotation[] qualifiersArray) {
+            Collection<? extends AdpSelectionStrategy> strategies =
+                concreteStrategies(c.get("oci.auth-strategies", String[].class)
```

Comment:
Because `strategies` is not a property of `auth`.  `auth-strategies` is its own "thing" that governs "which" `auth` is used.  Why the hyphens? Because apparently the development guidelines mandate their use.

Reply:
@klustria I'm going to resolve this conversation unless you have more to add here?

Reply:
Sure I'm fine with this